src/classes/Auth.php

```diff
@@ -134,6 +144,8 @@ public function needAuth(): bool
             'login.php',
             'LoginController.php',
             'metadata.php',
+            'mfa.php',
```

Comment:
we should avoid adding more single php files. maybe use login.php and LoginController instead

Reply:
My idea of having the MFA code in a separate files is that one could use MFA not only during login but maybe also somewhere else. E.g. if user changes the password or if (sys)admin does something delicate or maybe set a timeout after which one needs to reverify.
Why do you want to avoid the addition of more files?

Reply:
Because php files that end up in the URL means more maintenance, and less clean code. Look at experiments.php and database.php, they are nearly the same, and yet the code lives in different files. That's because of "history". But here we don't need to do the same mistake. Having the code tucked away in a controller/class doesn't stop one from using it anywhere (which is a good idea btw). I would really like to avoid having more direct access php files in web/.

Reply:
Less maintenance makes sense to me. I transplanted the logic into login.php and LoginController.php, as suggested.
For me it works very smoothly now. However, I did not test whether it interferes with SAML or authentication by web server.